File: paddler_client_python/paddler_client/embedding.py
from pydantic import BaseModel, ConfigDict

from paddler_client.embedding_normalization_method import (
    EmbeddingNormalizationMethod,
)
from paddler_client.pooling_type import PoolingType


class Embedding(BaseModel):
    model_config = ConfigDict(frozen=True)

    embedding: list[float]
    normalization_method: EmbeddingNormalizationMethod
    pooling_type: PoolingType
    source_document_id: str

File: paddler_client_javascript/src/schemas/Embedding.ts
import { z } from "zod";

import { EmbeddingNormalizationMethodSchema } from "./EmbeddingNormalizationMethod";
import { PoolingTypeSchema } from "./PoolingType";

export const EmbeddingSchema = z.object({
  embedding: z.array(z.number()),
  normalization_method: EmbeddingNormalizationMethodSchema,
  pooling_type: PoolingTypeSchema,
  source_document_id: z.string(),
});

export type Embedding = z.infer<typeof EmbeddingSchema>;
